src/mavm.rs

```diff
@@ -275,12 +275,317 @@ impl fmt::Display for CodePt {
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum BufferElem {
+    Leaf(Rc<RefCell<Vec<u8>>>),
+    Node(Rc<RefCell<Vec<Buffer>>>, u8),
+    Sparse(Rc<RefCell<Vec<usize>>>, Rc<RefCell<Vec<u8>>>, u8),
+}
```

Comment:
I am a bit confused on why we need `RefCell`s here, can you please explain what they are needed for?